### PR TITLE
AccountView.vala: Validate entire form

### PR DESCRIPTION
**Pull request by danrabbit**
* One entry = one validation method
* Cache the result of validating an entry
* When an entry changes, check to see if entries are valid and if so set the finish button's sensitivity

### CODE REVIEW

File: src/Views/AccountView.vala [hunk 1]
```diff
@@ -146,22 +163,28 @@ public class Installer.AccountView : AbstractInstallerView {
...
                 confirm_entry.set_icon_from_icon_name (Gtk.EntryIconPosition.SECONDARY, "dialog-error-symbolic");
                 confirm_entry.tooltip_text = _("Passwords do not match");
             } else {
                 confirm_entry.set_icon_from_icon_name (Gtk.EntryIconPosition.SECONDARY, "process-completed-symbolic");
                 confirm_entry.tooltip_text = null;
+                return true;
             }
         } else {
             confirm_entry.set_icon_from_icon_name (Gtk.EntryIconPosition.SECONDARY, null);
         }
```
Comment (donadigo) [inline src/Views/AccountView.vala]: Whitespace here.

File: src/Views/AccountView.vala [hunk 2]
```diff
@@ -182,6 +206,15 @@ public class Installer.AccountView : AbstractInstallerView {
             username_error_revealer.reveal_child = true;
             username_entry.set_icon_from_icon_name (Gtk.EntryIconPosition.SECONDARY, "dialog-error-symbolic");
         }
+        return false;
+    }
+
+    private void validate_form () {
```
Comment (donadigo) [inline src/Views/AccountView.vala]: Can we be more explicit here? Like calling this `update_finish_button`?

File: src/Views/AccountView.vala [hunk 3]
```diff
@@ -182,6 +206,15 @@ public class Installer.AccountView : AbstractInstallerView {
             username_error_revealer.reveal_child = true;
             username_entry.set_icon_from_icon_name (Gtk.EntryIconPosition.SECONDARY, "dialog-error-symbolic");
         }
```
Comment (donadigo) [inline src/Views/AccountView.vala]: Whitespace here.

File: src/Views/AccountView.vala [hunk 4]
```diff
@@ -108,14 +113,26 @@ public class Installer.AccountView : AbstractInstallerView {
             username_entry.text = username;
         });
 
-        username_entry.changed.connect (check_username);
-        pw_entry.changed.connect (check_password);
-        confirm_entry.changed.connect (check_password);
+        username_entry.changed.connect (() => {
+            valid_username = check_username ();
```
Comment (donadigo) [inline src/Views/AccountView.vala]: I'm not sure this is an improvement over the current method of validating. If we are validating anything inside the dialog we should explicitly update the finish button, in this case I think I would prefer to call `validate_form` inside `check_` and `confirm_` methods. The `valid_` fields should just be set inside those methods.

Comment (danrabbit) [reply src/Views/AccountView.vala]: That breaks down with checking and confirming the password which must be both be done if the password entry changes, but only confirming must be done if the confirm entry changes. If each method called `validate_form` then every time the password confirmation entry changed, we would validate the form twice.

**Comment by danrabbit** [comment]
In future revisions I'd like to think of ways we can separate the UI a bit more, but these validations are much more complicated than just "valid" and "invalid" with password for example being "valid" but being a "warning" and there not being a single error or warning string.

In the current state, I would just like to get this working so we can move forward with an MVP

**Pull request merged**